modules/bitcoind.nix

```diff
@@ -268,7 +268,7 @@ in {
           mkdir -m 0770 -p '${cfg.dataDir}/blocks'
         fi
         chown -R '${cfg.user}:${cfg.group}' '${cfg.dataDir}'
-        chmod -R g+rX '${cfg.dataDir}/blocks'
+        ${if cfg.dataDirReadableByGroup then "chmod -R g+rX '${cfg.dataDir}/blocks'" else "chmod -R g-rX '${cfg.dataDir}/blocks'"}
```

Comment:
We have to set permissions here because we need to make _existing_ blocks directories and their contents group-readable/unreadable depending on the `cfg.dataDirReadableByGroup` option. This is relevant for users switching `cfg.high-memory` on and off. If we only do it on creation we don't handle this case.

If we create both `cfg.dataDir` and `cfg.dataDir/blocks` with `systemd.tmpfiles.rules -d` we don't handle the blocks directory's contents. If we try to use `systemd.tmpfiles.rules -Z` to handle the blocks directory's contents, we can't handle directories and files differently like with `chmod g+X`

So I believe the way we do it now is best.

https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html

Reply:
Ok makes sense